cli/parse_pdfs.py

```diff
@@ -259,11 +260,22 @@ def parse_file(
             )
             return None
 
-        # TODO add error handling for azure api response and retry with large document
-        #  api
-        api_response: AnalyzeResult = azure_client.analyze_document_from_bytes(
-            doc_bytes=read_local_json_to_bytes(str(pdf_path)),
-        )
+        # TODO retry with large document method if default fails
+        try:
+            api_response: AnalyzeResult = azure_client.analyze_document_from_bytes(
+                doc_bytes=read_local_json_to_bytes(str(pdf_path)),
+            )
+        except ServiceRequestError as e:
```

Comment:
do we know that `ServiceRequestError` is always the error that you've detailed in the exception? Are there any properties of the error we can log?

Reply:
How about this? 

`        try:
            api_response: AnalyzeResult = azure_client.analyze_document_from_bytes(
                doc_bytes=read_local_json_to_bytes(str(pdf_path)),
            )
        except ServiceRequestError as e:
            _LOGGER.exception(
                "Failed to parse document with Azure API. This is most likely due to "
                "incorrect azure api credentials.", 
                extra={
                    "props": {
                        "document_id": input_task.document_id,
                        "error_message": str(e.message),
                    }
                },
            )
            return None
        except Exception as e:
            _LOGGER.exception(
                "Failed to parse document with Azure API.",
                extra={
                    "props": {
                        "document_id": input_task.document_id,
                        "error_message": str(e),
                    }
                },
            )
            return None`

Reply:
that looks great!